src/controllers/OtherReasonController.ts

```diff
@@ -5,33 +5,49 @@ import { SchemaValidator } from '../utils/validation/SchemaValidator';
 import { ValidationResult } from '../utils/validation/ValidationResult';
 import { OtherReason } from '../models/OtherReason';
 import { schema } from '../models/OtherReason.schema';
-import { RedisService } from '../services/RedisService';
 import { OK, UNPROCESSABLE_ENTITY } from 'http-status-codes';
+import { Request } from 'express';
+import { Cookie } from 'ch-node-session/lib/session/model/Cookie';
+import { EitherUtils, SessionMiddleware, SessionStore } from 'ch-node-session';
+import { AuthMiddleware } from '../middleware/AuthMiddleware';
 
 const sessionKey = 'session::other-reason';
 
-@controller(OTHER_REASON_PAGE_URI)
+@controller(OTHER_REASON_PAGE_URI, SessionMiddleware, AuthMiddleware)
 export class OtherReasonController extends BaseHttpController {
-    constructor(@inject(RedisService) private readonly redisService: RedisService) {
+    constructor(@inject(SessionStore) private readonly store: SessionStore) {
```

Comment:
can we call this `sessionStore`? as there is a `store` method on the SessionStore class